automation/functions/script.ps1

```diff
@@ -1,26 +1,38 @@
-param(
+ param(
     [Parameter(Mandatory = $True)]
     [string]
-    $resourceGroupName,
+    $SubscriptionId,
 
     [Parameter(Mandatory = $True)]
     [string]
-    $location
+    $TenantId,
+
+    [Parameter(Mandatory = $True)]
+    [string]
+    $ServicePrincipalId,
+
+    [Parameter(Mandatory = $True)]
+    [string]
+    $ServicePrincipalPassword
 )
 
-#sign into azure account 
-Connect-AzAccount
```

Comment:
This will pop up a browser window. You should remove this line - you already have the login kind of correct on line 23.